Hotfix for actor subsystem initialization in 1.7 (and possibly others) (#4622)
dapr/dapr: docs/release_notes/v1.7.3.md

# Dapr 1.7.3

### Fixes incorrect actor state-store configuration behavior

#### Problem

> Omitting a state store configuration will cause the actor subsystem to fail to initialize. This is the correct behavior for services providing actors but this not for services acting as consumers / clients of actors that other services provide. A service acting purely as a consumer does not need state storage and should initiazlize correctly in order to invoke actors elsewhere. 

Reviewer: "state storage" -> "actor state storage" (both occurrences)

"but this not for" -> "but not for"

"A service acting purely as a consumer" -> client instead of consumer

#### Root cause

> Absence of a state store causes the actor subsystem to throw an error, regardless of whether or not the service is providing hosted actors.

Reviewer: Root Cause: All Actor APIs return an error if no actor state store is defined, regardless of where actors are registered.

FYI SDKs use the term "register" for telling a sidecar to host particular actors. So this may be less confusing to use in language.

#### Solution

> Thix fix changes the actor runtime initialization logic such that, when there is no state store available, the actor subsystem will successfully initialize for consumers but will fail for services with hosted actors. 

Reviewer: state store -> actor state store
consumer -> client

# Dapr 1.7.3

### Fixes incorrect actor state-store configuration behavior

#### Problem

> The Actor API fails to initialize without an actor state store component being provided. This is the correct behavior for services that register actors, as they require state storage, but clients that invoke actors should operate with or without an actor state store. 

Reviewer: ```suggestion
All Actor APIs return errors without an actor state store component being provided. This is the correct behavior for services that register actors, as they require state storage, but clients that invoke actors should operate with or without an actor state store. 
```
